File: 연습문제/프로그래머스_기능개발.cpp
#include <bits/stdc++.h>

using namespace std;

vector<int> solution(vector<int> progresses, vector<int> speeds) {
	vector<int> answer;
	int idx = 0;

	while (idx < progresses.size()) {
		int cnt = 0;
		
		for (int i = 0; i < progresses.size(); i++)
			progresses[i] += speeds[i];

		while (progresses[idx] >= 100 && idx < progresses.size()) {
			idx++;
			cnt++;
		}

		if (cnt != 0)
			answer.push_back(cnt);
		
	}

	return answer;
}

int main(void) {
	vector<int> progresses = { 93, 30, 55 };
	// {95, 90, 99, 99, 80, 99}
	vector<int> speeds = { 1, 30, 5 };
	// {1, 1, 1, 1, 1, 1}

	solution(progresses, speeds);

	
	return 0;
}

File: 스택/프로그래머스_기능개발.py
def solution(progresses, speeds):
    answer = []

    while progresses:
        cnt=0
        while (len(progresses)!=0 and progresses[0]>=100):
            cnt+=1
            progresses.pop(0)
            speeds.pop(0)
        if cnt:
            answer.append(cnt)
        
        progresses = [progresses[i] + speeds[i] for i in range(len(progresses))]
        
    return answer

solution([93,30,55], [1,30,5])